### PR TITLE
Moving ec2 (17) resources to v1beta1 version #65

### LINKED ISSUE #65: Moving ec2 (17) resources to v1beta1 version
Moving 17 resources in the `ec2` group:
- [ ] [`aws_default_subnet`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/default_subnet)
- [ ] [`aws_default_vpc`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/default_vpc)
- [ ] [`aws_default_vpc_dhcp_options`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/default_vpc_dhcp_options)
- [ ] [`aws_ebs_default_kms_key`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ebs_default_kms_key)
- [ ] [`aws_ebs_encryption_by_default`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ebs_encryption_by_default)
- [ ] [`aws_ebs_snapshot_copy`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ebs_snapshot_copy)
- [ ] [`aws_ebs_snapshot_import`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ebs_snapshot_import)
- [ ] [`aws_ec2_availability_zone_group`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ec2_availability_zone_group)
- [ ] [`aws_ec2_capacity_reservation`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ec2_capacity_reservation)
- [ ] [`aws_ec2_carrier_gateway`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ec2_carrier_gateway)
- [ ] [`aws_ec2_host`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ec2_host)
- [ ] [`aws_ec2_network_insights_path`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ec2_network_insights_path)
- [ ] [`aws_ec2_serial_console_access`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ec2_serial_console_access)
- [ ] [`aws_ec2_subnet_cidr_reservation`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ec2_subnet_cidr_reservation)
- [ ] [`aws_ec2_traffic_mirror_filter`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ec2_traffic_mirror_filter)
- [ ] [`aws_ec2_traffic_mirror_filter_rule`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ec2_traffic_mirror_filter_rule)
- [ ] [`aws_ec2_transit_gateway_connect`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ec2_transit_gateway_connect)

### PR DESCRIPTION
**Pull request by dverveiko**
### Description of your changes

Fixes #65
- [x] [`aws_default_subnet`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/default_subnet)
- [x] [`aws_default_vpc`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/default_vpc)
- [x] [`aws_default_vpc_dhcp_options`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/default_vpc_dhcp_options)
- [x] [`aws_ebs_default_kms_key`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ebs_default_kms_key)
- [x] [`aws_ebs_encryption_by_default`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ebs_encryption_by_default)
- [x] [`aws_ebs_snapshot_copy`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ebs_snapshot_copy)
- [x] [`aws_ebs_snapshot_import`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ebs_snapshot_import)
- [x] [`aws_ec2_availability_zone_group`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ec2_availability_zone_group)
- [x] [`aws_ec2_capacity_reservation`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ec2_capacity_reservation)
- [x] [`aws_ec2_carrier_gateway`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ec2_carrier_gateway)
- [x] [`aws_ec2_host`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ec2_host)
- [x] [`aws_ec2_network_insights_path`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ec2_network_insights_path)
- [x] [`aws_ec2_serial_console_access`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ec2_serial_console_access)
- [x] [`aws_ec2_subnet_cidr_reservation`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ec2_subnet_cidr_reservation)
- [x] [`aws_ec2_traffic_mirror_filter`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ec2_traffic_mirror_filter)
- [x] [`aws_ec2_traffic_mirror_filter_rule`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ec2_traffic_mirror_filter_rule)
- [x] [`aws_ec2_transit_gateway_connect`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ec2_transit_gateway_connect)

I have:

- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested
Manually

### CODE REVIEW

File: examples/ec2/availabilityzonegroup.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  annotations:
+    meta.upbound.io/example-id: ec2/v1beta1/availabilityzonegroup
+  labels:
+    testing.upbound.io/example-name: example
+  name: us-west-2-den-1
+spec:
+  forProvider:
+    optInStatus: opted-in
+    region: us-west-2
+
```
Comment (turkenh) [inline examples/ec2/availabilityzonegroup.yaml:13]: Please clean these empty lines and separators from the examples. They cause build to fail with 

```
up: error: failed to build package: failed to parse examples: error converting YAML to JSON: yaml: line 2: did not find expected node content
```

**Review by turkenh — COMMENTED** [review]
Please also rebase your change on top of latest main.

**Comment by turkenh** [comment]
Can you add _test-examples_ comment to see if it works for you?

**Comment by ulucinar** [comment]
Hi @dverveiko, could you please add a comment like the following:
```
/test-examples="examples/ec2/availabilityzonegroup.yaml,examples/ec2/capacityreservation.yaml,examples/ec2/carriergateway.yaml,examples/ec2/defaultsubnet.yaml,examples/ec2/defaultvpc.yaml,examples/ec2/defaultvpcdhcpoptions.yaml,examples/ec2/ebsdefaultkmskey.yaml,examples/ec2/ebsencryptionbydefault.yaml,examples/ec2/ebssnapshotcopy.yaml,examples/ec2/ebssnapshotimport.yaml,examples/ec2/host.yaml,examples/ec2/networkinsightspath.yaml,examples/ec2/serialconsoleaccess.yaml,examples/ec2/subnetcidrreservation.yaml,examples/ec2/trafficmirrorfilter.yaml,examples/ec2/trafficmirrorfilterrule.yaml,examples/ec2/transitgatewayconnect.yaml"
```

**Comment by dverveiko** [comment]
/test-examples="examples/ec2/availabilityzonegroup.yaml,examples/ec2/capacityreservation.yaml,examples/ec2/carriergateway.yaml,examples/ec2/defaultsubnet.yaml,examples/ec2/defaultvpc.yaml,examples/ec2/defaultvpcdhcpoptions.yaml,examples/ec2/ebsdefaultkmskey.yaml,examples/ec2/ebsencryptionbydefault.yaml,examples/ec2/ebssnapshotcopy.yaml,examples/ec2/ebssnapshotimport.yaml,examples/ec2/host.yaml,examples/ec2/networkinsightspath.yaml,examples/ec2/serialconsoleaccess.yaml,examples/ec2/subnetcidrreservation.yaml,examples/ec2/trafficmirrorfilter.yaml,examples/ec2/trafficmirrorfilterrule.yaml,examples/ec2/transitgatewayconnect.yaml"

**Comment by dverveiko** [comment]
/test-examples="examples/ec2/availabilityzonegroup.yaml,examples/ec2/capacityreservation.yaml,examples/ec2/carriergateway.yaml,examples/ec2/defaultsubnet.yaml,examples/ec2/defaultvpc.yaml,examples/ec2/defaultvpcdhcpoptions.yaml,examples/ec2/ebsdefaultkmskey.yaml,examples/ec2/ebsencryptionbydefault.yaml,examples/ec2/ebssnapshotcopy.yaml,examples/ec2/ebssnapshotimport.yaml,examples/ec2/host.yaml,examples/ec2/networkinsightspath.yaml,examples/ec2/serialconsoleaccess.yaml,examples/ec2/subnetcidrreservation.yaml,examples/ec2/trafficmirrorfilter.yaml,examples/ec2/trafficmirrorfilterrule.yaml,examples/ec2/transitgatewayconnect.yaml"

**Comment by dverveiko** [comment]
/test-examples="examples/ec2/availabilityzonegroup.yaml,examples/ec2/capacityreservation.yaml,examples/ec2/carriergateway.yaml,examples/ec2/defaultsubnet.yaml,examples/ec2/defaultvpc.yaml,examples/ec2/defaultvpcdhcpoptions.yaml,examples/ec2/ebsdefaultkmskey.yaml,examples/ec2/ebsencryptionbydefault.yaml,examples/ec2/ebssnapshotcopy.yaml,examples/ec2/ebssnapshotimport.yaml,examples/ec2/host.yaml,examples/ec2/networkinsightspath.yaml,examples/ec2/serialconsoleaccess.yaml,examples/ec2/subnetcidrreservation.yaml,examples/ec2/trafficmirrorfilter.yaml,examples/ec2/trafficmirrorfilterrule.yaml,examples/ec2/transitgatewayconnect.yaml"

**Comment by dverveiko** [comment]
/test-examples="examples/ec2/availabilityzonegroup.yaml,examples/ec2/capacityreservation.yaml,examples/ec2/carriergateway.yaml,examples/ec2/defaultsubnet.yaml,examples/ec2/defaultvpc.yaml,examples/ec2/defaultvpcdhcpoptions.yaml,examples/ec2/ebsdefaultkmskey.yaml,examples/ec2/ebsencryptionbydefault.yaml,examples/ec2/ebssnapshotcopy.yaml,examples/ec2/ebssnapshotimport.yaml,examples/ec2/host.yaml,examples/ec2/networkinsightspath.yaml,examples/ec2/serialconsoleaccess.yaml,examples/ec2/subnetcidrreservation.yaml,examples/ec2/trafficmirrorfilter.yaml,examples/ec2/trafficmirrorfilterrule.yaml,examples/ec2/transitgatewayconnect.yaml"

**Comment by dverveiko** [comment]
/test-examples="examples/ec2/availabilityzonegroup.yaml,examples/ec2/capacityreservation.yaml,examples/ec2/carriergateway.yaml,examples/ec2/defaultsubnet.yaml,examples/ec2/defaultvpc.yaml,examples/ec2/defaultvpcdhcpoptions.yaml,examples/ec2/ebsdefaultkmskey.yaml,examples/ec2/ebsencryptionbydefault.yaml,examples/ec2/ebssnapshotcopy.yaml,examples/ec2/ebssnapshotimport.yaml,examples/ec2/host.yaml,examples/ec2/networkinsightspath.yaml,examples/ec2/serialconsoleaccess.yaml,examples/ec2/subnetcidrreservation.yaml,examples/ec2/trafficmirrorfilter.yaml,examples/ec2/trafficmirrorfilterrule.yaml,examples/ec2/transitgatewayconnect.yaml"

**Comment by dverveiko** [comment]
/test-examples="examples/ec2/ebssnapshotimport.yaml"

**Comment by dverveiko** [comment]
/test-examples="examples/ec2/ebssnapshotimport.yaml"

**Comment by dverveiko** [comment]
/test-examples="examples/ec2/transitgatewayconnect.yaml"

**Comment by dverveiko** [comment]
/test-examples="examples/ec2/defaultsubnet.yaml"

**Comment by dverveiko** [comment]
/test-examples="examples/ec2/networkinsightspath.yaml"

**Review by ezgidemirel — APPROVED** [review]
Thanks a lot @dverveiko!!

**Pull request merged**